MFVNeutralino/test/TrackMover/robust_uncerts_mm.py

```diff
@@ -0,0 +1,360 @@
+
+def scaledTOC(sig_curve, data_curve, sim_curve):
+    s_curve = ROOT.TH1D("scaled_curve", "", 80, 0, 80)
+    fout = ROOT.TFile("scaledtoc.root", "recreate")
+    for b in range(0,80):
+        if (sim_curve.GetBinContent(b) > 0):
+            s_curve.SetBinContent(b, sig_curve.GetBinContent(b)*data_curve.GetBinContent(b)/sim_curve.GetBinContent(b))
+            s_curve.SetBinError(b, sig_curve.GetBinError(b)*data_curve.GetBinContent(b)/sim_curve.GetBinContent(b))
+    s_curve.Write()
+    return s_curve
+
+def cutZero(original, name):
+    curve = original.Clone()
+    for b in range(0,80):
+        curve.SetBinContent(1, 0)
+        curve.SetBinError(1, 0)
+    return curve
+
+#############################################################################################
+
+def shiftDIST(den, sint, fr):
+    s_den = den.Clone() #ROOT.TH1D("placeholder", "", 80, 0, 80)
+
+    fr = 1.0-fr
+
+    for b in range(0,den.GetNbinsX()):
+        s_den.SetBinContent(b, fr*den.GetBinContent(b+sint) + (1-fr)*den.GetBinContent(b+1+sint))
+        s_den.SetBinError(b, np.hypot(fr*den.GetBinError(b+sint), (1-fr)*den.GetBinError(b+1+sint)))
+
+    return s_den
+
+################################################################################################
+
+def assessMCToDataUncerts(slide_uncerts, scale_uncerts, toc_uncerts, tkscl_uncerts, mc_unc, stat_unc):
+
+    sys_dist_val = max(abs(slide_uncerts), abs(scale_uncerts))
+    #print("slide n scale distr", slide_uncerts, scale_uncerts)
+    total = np.sqrt (sys_dist_val**2 + toc_uncerts**2 + 0.0**2 + stat_unc**2)
+
+    print( " 1-vtx Unc. by SF_{nclsedtks,non} : %.3f (sys_distr) +/- %.3f (sys_scale_toc) +/- %.3f (sig_stat) +/- %.3f (sys_reweight) (+/- %.3f (sys_tkrescl)?) : %.3f %% " % (sys_dist_val, toc_uncerts, stat_unc, 0.0, tkscl_uncerts, total) )
+
+    return total
+
+################################################################################################
+
+def calcTocShiftUncert(low, cent, hi):
+
+    outRmsVals = []
+
+    for i in range(0, len(low)):
+        rms =  np.sqrt( ((cent[i] - low[i])**2 + (cent[i]-hi[i])**2)/2 )
+        rms = round(rms, 3)
+        outRmsVals.append(rms)
+
+    return outRmsVals
+
+
+
+################################################################################################
+
+
+# Initialize stuff:
+
+year = '2017'
+doShift  = True
+reweight = True
+#toc_shift = 0.0   # How much to move the turn-on curve by
+#shift_fr  = 0.0   # How much to slide the closeseedtk dist by (decimal part)
+#shift_val = 0     # How much to slide the closeseedtk dist by (integer part)
+
+masses = ['55'] # ['15','40','55']
+ctaus       = ['1000'] #['1000','3000','30000'] 
+psd_methods = ['none', 'slide_distr', 'scale_distr', 'scale_toc', 'trackrescl']
+
+"""
+if year == '2018': 
+    shift_val = 2 #FIXME
+    shift_fr  = 0.0 #FIXME
+    toc_shift = 1.0 #FIXME 
+if year == '2017': 
+    shift_val = 0 #FIXME
+    shift_fr  = 0.037 #FIXME 
+    toc_shift = 1.0 #FIXME
+"""
```

Comment:
Delete?

Reply:
Please ignore this script. Remove it. Done.